lib/chart/internal/plot.ex

```diff
@@ -1,24 +1,85 @@
 defmodule Chart.Internal.Plot do
   @moduledoc false
 
+  alias Chart.Internal.AxisLine.Helpers
+
+  import Chart.Internal.Guards,
+    only: [is_nonnegative_number: 1, is_positive_numbers: 2, is_numbers: 2]
+
+  @self_key :plot
+
   def new() do
     %{
+      axis: [],
```

Comment:
Proc je tu axis? A proc je to list? Flat je lepsi nez nested.

Reply:
Nevim co je flat, ale list `:axis` obsahuje vsechny definovane osy. Je to  z toho duvodu, ze kdyz se treba zavola `Plot.set_size({600, 500})` tak aby se zmenily `MajorTicks...` pozice apod.

Reply:
a ktera osa je x a ktera y, pripadne z?

Reply:
co v tom listu je?

Reply:
Flat je plocha struktura. Zanoreni prinasi problemy.

Reply:
> Nevim co je flat, ale list `:axis` obsahuje vsechny definovane osy. Je to z toho duvodu, ze kdyz se treba zavola `Plot.set_size({600, 500})` tak aby se zmenily `MajorTicks...` pozice apod.

Proc by toho neslo dosahnout i kdyz jsou axis v rootu struktury?

Reply:
Myslim, ze tady neni v tuto chvili potreba vedet ktera osa je ktera. Spis kolik jich je.
V listu jsou uzivatelem definovane osy (viz cast `Settings.new() |> ... |> axis(:x_axis) |> axis(:y_axis) |> ...`). Tedy 
```
axis: [:x_axis, :y_axis]
```

Reply:
No takhle, puvodne jsem si rikal, ze `Figure` muzem mit vic jak jeden plot, napr 
![subplot](https://user-images.githubusercontent.com/13625279/90016738-c1ab4700-dcaa-11ea-9456-14568d4d99c2.png)
nebo
![subplot2](https://user-images.githubusercontent.com/13625279/90016797-db4c8e80-dcaa-11ea-9be5-913c55e8a4ae.png)


pak by to zanoreni davalo vetsi smysl, nez drzet vse na stejne urovni.

Reply:
Tuto informaci, preci mas uz v kodu, tedy v `Chart.SomeChart.Settings` jako  `axis(:x_axis)` a `axis(:y_axis)` a tedy tento modul vi, ktere osy tam jsou, pak by i zde meli byt vsechny funkce, ktere provadi operace nad provazanymi daty. Co se stane, pokud by ti nejakou zahadou zmizela osa z rootu nebo z `plot.axis`, tak budes renderovat napul validni graf.

Reply:
> No takhle, puvodne jsem si rikal, ze `Figure` muzem mit vic jak jeden plot, napr
> ![subplot](https://user-images.githubusercontent.com/13625279/90016738-c1ab4700-dcaa-11ea-9456-14568d4d99c2.png)
> nebo
> ![subplot2](https://user-images.githubusercontent.com/13625279/90016797-db4c8e80-dcaa-11ea-9be5-913c55e8a4ae.png)
> 
> pak by to zanoreni davalo vetsi smysl, nez drzet vse na stejne urovni.

To je zbytecne. To si muze uzivatel zaridit sam.

Reply:
Teda jestli te chapu spravne tak by napr v modulu `Plot` pri zmene size to melo byt nejak takto? (tech Helpers se tedka zbavuji, v dalsim commitu uz nebudou)
```
  def set_position(settings, {x, y} = position)
      when is_map(settings) and is_number(x) and is_number(y) do
    axis = settings.plot.axis

    settings
    |> put_in(
      [@self_key, :position],
      validate_position(position, settings.figure.viewbox, settings.plot.size)
    )
    |> MajorTicks.set_positions(:x_axis)
    |> MajorTicks.set_positions(:y_axis)
    |> ...
  end
```

Reply:
JJ, vyhnout se duplicitnimu popisu jednoho stavu.

Reply:
Tomuto jsem se prave chtel vyhnout. Muze se stat ze prijde nejaky contributor co si bude chtit udelat vlastni graf a rekne si ze nechce mit `:x_axis` ale napr `:axis1` a pak mu to prestane fungovat. Resp si muze pridat treba dalsi `axis(:z_axis)` a taky pak bude hledat kde je chyba. S tim `:axis` listem by neresil nic, vse by mu fungovalo.

Reply:
Pokud bude vytvaret vlastni graf, pak bude vytvaret neco jako `SomeChart`, `SomeChart.Settings`, ... a pak bude vytvaret nastaveni toho grafu, kde muze pouzit misto `:x_axis` cokoliv a jelikoz ty internaly nemaji vazbu na tento klic, tak se nic nerozbije a on vi jake klice pouzil a tim je bude i volat. Pokud prijde konzument jeho grafu, tak tam problem byt muze, ale to uz musi byt uvedene v dokumentaci.

Reply:
Stejne si myslim, ze bude existovat api pro externi uzivatele nekde v `SomeChart` a internich veci se vubec uzivatel nedotkne, pokud si nebude opravdu jisty, ze to ptrebuje. Tedy bude existovat napr. `SomeChar.set_label("Muj graf")`.

Reply:
> kde muze pouzit misto :x_axis cokoliv a jelikoz ty internaly nemaji vazbu na tento klic

Praveze timto tu vazbu mit budou, fce `set_position` je implementovana v `Char.Internal.Plot` modulu. Takze si bude muset sahnout i do tohoto modulu a prepsat si to aby mu to sedelo. 

> Tedy bude existovat napr.

Souhlas, take bych to tak videl... 

Jinak to poresime zitra...